sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchangeExec.scala

```diff
@@ -299,7 +300,8 @@ object ShuffleExchangeExec {
     def getPartitionKeyExtractor(): InternalRow => Any = newPartitioning match {
       case RoundRobinPartitioning(numPartitions) =>
         // Distributes elements evenly across output partitions, starting from a random partition.
-        var position = new Random(TaskContext.get().partitionId()).nextInt(numPartitions)
```

Comment:
Sorry I may miss something. The original code should already produce different starting positions for different mapper tasks?

Reply:
OK I tried `(1 to 200).foreach(partitionId => print(new Random(partitionId).nextInt(32) + " "))` and the result is very counterintuitive. A small change for the seed does not change the random result.

Can we add some comments to explain why we add `hashing.byteswap32`?

Reply:
This was fixed in SPARK-21782 for RDD - looks like the sql version did not leverage it.

Reply:
@mridulm wow, good findings. I didn't realize there was a similar issue.